frontend/app/src/components/history/TransactionEventForm.vue

```diff
@@ -223,326 +224,280 @@ import { bigNumberifyFromRef, One, Zero } from '@/utils/bignumbers';
 import { convertFromTimestamp, convertToTimestamp } from '@/utils/date';
 import { getEventTypeData } from '@/utils/history';
 
-const TransactionEventForm = defineComponent({
-  name: 'TransactionEventForm',
-  components: { ValueAccuracyHint, LocationSelector },
-  props: {
-    value: { required: false, type: Boolean, default: false },
-    edit: {
-      required: false,
-      type: Object as PropType<EthTransactionEvent>,
-      default: null
-    },
-    transaction: {
-      required: false,
-      type: Object as PropType<EthTransactionEntry>,
-      default: null
-    },
-    saveData: {
-      required: true,
-      type: Function as PropType<
-        (
-          event: NewEthTransactionEvent | EthTransactionEventEntry
-        ) => Promise<ActionStatus>
-      >
-    }
+const props = defineProps({
+  value: { required: false, type: Boolean, default: false },
+  edit: {
+    required: false,
+    type: Object as PropType<EthTransactionEvent>,
+    default: null
   },
-  emits: ['input'],
-  setup(props, { emit }) {
-    const { t } = useI18n();
-    const { edit, transaction, saveData } = toRefs(props);
+  transaction: {
+    required: false,
+    type: Object as PropType<EthTransactionEntry>,
+    default: null
+  },
+  saveData: {
+    required: true,
+    type: Function as PropType<
+      (
+        event: NewEthTransactionEvent | EthTransactionEventEntry
+      ) => Promise<ActionStatus>
+    >
+  }
+});
 
-    const input = (valid: boolean) => emit('input', valid);
+const emit = defineEmits(['input']);
```

Comment:
It would be better to properly type the emitted events for type-checking.

It would be better to avoid the array syntax and use the proper typed syntax `defineEmits<{(e: 'input', valid: boolean):void}>()`